scripts/publish-npm.js

```diff
@@ -46,133 +46,133 @@ const fs = require('fs');
-const isCommitly = nightlyBuild || dryRunBuild;
-
-const buildType = releaseBuild
-  ? 'release'
-  : nightlyBuild
-  ? 'nightly'
-  : 'dry-run';
-
-// 34c034298dc9cad5a4553964a5a324450fda0385
-const currentCommit = getCurrentCommit();
-const shortCommit = currentCommit.slice(0, 9);
-
-const rawVersion =
-  // 0.0.0 triggers issues with cocoapods for codegen when building template project.
-  dryRunBuild
-    ? '1000.0.0'
-    : // For nightly we continue to use 0.0.0 for clarity for npm
-    nightlyBuild
-    ? '0.0.0'
-    : // For pre-release and stable releases, we use the git tag of the version we're releasing (set in bump-oss-version)
-      buildTag;
-
-let version,
-  major,
-  minor,
-  prerelease = null;
-try {
-  ({version, major, minor, prerelease} = parseVersion(rawVersion, buildType));
-} catch (e) {
-  echo(e.message);
-  exit(1);
-}
+function getNpmInfo(buildType) {
+  const currentCommit = getCurrentCommit();
+  const shortCommit = currentCommit.slice(0, 9);
 
-let releaseVersion;
-if (dryRunBuild) {
-  releaseVersion = `${version}-${shortCommit}`;
-} else if (nightlyBuild) {
-  // 2021-09-28T05:38:40.669Z -> 20210928-0538
-  const dateIdentifier = new Date()
-    .toISOString()
-    .slice(0, -8)
-    .replace(/[-:]/g, '')
-    .replace(/[T]/g, '-');
-  releaseVersion = `${version}-${dateIdentifier}-${shortCommit}`;
-} else {
-  releaseVersion = version;
-}
+  if (buildType === 'dry-run') {
+    return {
+      version: `1000.0.0-${shortCommit}`,
+      tag: null, // We never end up publishing this
+    };
+  }
 
-// Bump version number in various files (package.json, gradle.properties etc)
-// For stable, pre-release releases, we rely on CircleCI job `prepare_package_for_release` to handle this
-if (isCommitly) {
-  if (
-    exec(
-      `node scripts/set-rn-version.js --to-version ${releaseVersion} --build-type ${buildType}`,
-    ).code
-  ) {
-    echo(`Failed to set version number to ${releaseVersion}`);
-    exit(1);
+  if (buildType === 'nightly') {
+    const dateIdentifier = new Date()
+      .toISOString()
+      .slice(0, -8)
+      .replace(/[-:]/g, '')
+      .replace(/[T]/g, '-');
+    return {
+      version: `0.0.0-${dateIdentifier}-${shortCommit}`,
+      tag: 'nightly',
+    };
   }
+
+  const {version, major, minor, prerelease} = parseVersion(
+    process.env.CIRCLE_TAG,
+    buildType,
+  );
+
+  // See if releaser indicated that this version should be tagged "latest"
+  // Set in `bump-oss-version`
+  const isLatest = exitIfNotOnGit(
+    () => isTaggedLatest(currentCommit),
+    'Not in git. We do not want to publish anything',
+  );
+
+  const releaseBranchTag = `${major}.${minor}-stable`;
+
+  // npm will automatically tag the version as `latest` if no tag is set when we publish
+  // To prevent this, use `releaseBranchTag` when we don't want that (ex. releasing a patch on older release)
+  const tag =
+    prerelease != null ? 'next' : isLatest ? 'latest' : releaseBranchTag;
```

Comment:
It does?! **takes notes 🗒️**

Reply:
IIRC! stack overflow to back me up: https://stackoverflow.com/questions/68230173/publishing-a-previous-major-version-to-npm-with-no-tag-and-avoiding-latest